tests/voq/test_voq_init.py

```diff
@@ -0,0 +1,434 @@
+"""Test initialization of VoQ objects, switch, system ports, router interfaces, neighbors, inband port."""
+import json
+import logging
+import pytest
+from tests.common.helpers.assertions import pytest_assert
+
+from tests.common.helpers.redis import AsicDbCli, RedisKeyNotFound
+from tests.common.errors import RunAnsibleModuleFail
+from voq_helpers import check_local_neighbor, check_voq_remote_neighbor, get_sonic_mac, get_neighbor_mac
+from voq_helpers import check_local_neighbor_asicdb, get_device_system_ports, get_inband_info, get_port_by_ip
+from voq_helpers import check_rif_on_sup, check_voq_neighbor_on_sup, find_system_port
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+
+@pytest.fixture(scope="module", autouse=True)
+def chassis_facts(duthosts):
+    """
+    Fixture to add some items to host facts from inventory file.
+    """
+    for a_host in duthosts.nodes:
+
+        if len(duthosts.supervisor_nodes) > 0:
+            out = a_host.command("cat /etc/sonic/card_details.json")
+            card_details = json.loads(out['stdout'])
+            if 'slot_num' in card_details:
+                a_host.facts['slot_num'] = card_details['slot_num']
+
+
+@pytest.fixture(scope="module")
+def nbrhosts_facts(nbrhosts):
+    nbrhosts_facts = {}
+    for a_vm in nbrhosts:
+        try:
+            vm_facts = nbrhosts[a_vm]['host'].eos_facts()
+        except RunAnsibleModuleFail:
+            logging.error("VM: %s is down, skipping config fetching.", a_vm)
```

Comment:
It would be better to initialize a logger than directly using the logging module.

Reply:
Fixed in 39f765f